01_oop_and_ood/src/Point.ts

```diff
@@ -0,0 +1,47 @@
+export class Point {
+    x: number;
+    y: number;
+
+    constructor()
+    constructor(x: number, y: number)
+    constructor(...args: any[]) {
+        if (!args.length) { 
+            this.x = 0; 
+            this.y = 0; 
+        } else if (typeof args[0] === 'number' && typeof args[1] === 'number') { 
+            this.x = args[0]; 
+            this.y = args[1]; 
+        } else { 
+            throw new Error('Invalid arguments.') 
+        } 
+    }
+
+    toString() {
+        return `(${this.x}, ${this.y})`;;
+    }
+
+    distance(): number;
+    distance(other: Point): number;
+    distance(x: number, y: number): number;
+    distance(...args: any[]): number {
+        if(!args.length) {
```

Comment:
same here. You have to work with zero arg, one arg, or two args.

Reply:
done